intezer_sdk/analyses_history.py

```diff
@@ -0,0 +1,159 @@
+                                   verdicts: List[str] = None,
+                                   limit: int = 100,
+                                   offset: int = 0
+                                   ):
+        """
+        Query for endpoint analyses history.
+
+        :param start_date: Date to query from.
+        :param end_date: Date to query until.
+        :param aggregate_view: Should the result be aggregated by latest
+        hash/url/computer.
+        :param sources: Filter the analyses by its source.
+        :param verdicts: Filter by the analysis's verdict
+        :param limit: Number of analyses returned by the query.
+        :param offset: Number of analyses to skips the before beginning to
+        return the analyses.
+        :return: all endpoint history analyses.
+        """
+        data = self._data_analyses_history(
+            start_date, end_date, aggregate_view, sources, verdicts, limit, offset
+        )
+        return Results('/endpoint-analyses/history', self.api, data)
+
+    def url_analyses_history(self, *,
+                             start_date: int,
+                             end_date: int,
+                             sources: List[str] = None,
+                             verdicts: List[str] = None,
+                             sub_verdicts: List[str] = None,
+                             did_download_file: bool = None,
+                             submitted_url: str = None,
+                             aggregate_view: bool = False,
+                             limit: int = 100,
+                             offset: int = 0
+                             ):
+        """
+        Query for url analyses history.
+
+        :param start_date: Date to query from.
+        :param end_date: Date to query until.
+        :param sources: Filter the analyses by its source.
+        :param verdicts: Filter by the analysis's verdict
+        :param sub_verdicts: Filter by the analysis's verdict
+        :param did_download_file: Should the result be aggregated by latest
+        hash/url/computer.
+        :param submitted_url: Filter by specific url
+        :param aggregate_view: Should the result be aggregated by latest
+        hash/url/computer.
+        :param limit: Number of analyses returned by the query.
+        :param offset: Number of analyses to skips the before beginning to
+        return the analyses.
+        :return: All url history analyses.
+        """
+        data = self._data_analyses_history(
+            start_date, end_date, aggregate_view, sources, verdicts, limit, offset
+        )
+
+        if did_download_file:
+            data["did_download_file"] = did_download_file
+        if submitted_url:
+            data["submitted_url"] = submitted_url
+        if sub_verdicts:
+            data["sub_verdicts"] = sub_verdicts
+
+        return Results('/url-analyses/history', self.api, data)
+
+    @staticmethod
+    def _data_analyses_history(*,
```

Comment:
I think in this method we don't need to put default values we already have them in the public method that use this private method

Reply:
I think for future changes it just help understand what I expect about the params, but it also ok without it so I changed it.